infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -897,35 +912,23 @@ private LogEntry getLogEntry(long address, LogData entry) {
     }
 
     /**
-     * Attempts to write a buffer to a file channel, if write fails with an
-     * IOException then the channel pointer is moved back to its original position
-     * before the write
+     * Attempts to write a buffer to a file channel.
      *
      * @param channel the channel to write to
      * @param buf     the buffer to write
      * @throws IOException IO exception
      */
-    private static void safeWrite(FileChannel channel, ByteBuffer buf) throws IOException {
-        long prev = channel.position();
-        try {
-            do {
-                channel.write(buf);
-            } while (buf.hasRemaining());
+    private void writeByteBuffer(FileChannel channel, ByteBuffer buf) throws IOException {
+        int bufSize = buf.remaining();
 
-        } catch (IOException e) {
-            // Write failed restore the channels position, so the subsequent writes
-            // can overwrite the failed write.
-
-            // Note that after rewinding the channel pointer, it is important to truncate
-            // any bytes that were written. This is required to avoid an ambiguous case
-            // where a subsequent write (after a failed write) succeeds but writes less
-            // bytes than the partially written buffer. In that case, the log unit can't
-            // determine if the bytes correspond to a partially written buffer that needs
-            // to be ignored, or if the bytes correspond to a corrupted metadata field.
-            channel.position(prev);
-            channel.truncate(prev);
-            channel.force(true);
-            throw e;
+        logSizeQuota.updateAndGet(quota -> {
```

Comment:
I believe this step should be after `while (buf.hasRemaining()) {`

There could be a case when you updated resourceQuota but the write is failed

Reply:
If its moved after the write, then writes will only fail after they have been written. 
Logically it should work like this:
reserve quota
do operation
if operation fails return the reserved quota

But in our case, IOExceptions are bubbled all the way up, so after such exception we will need to reinitialize the LogUnit which will reinitialize the quota.